internal/webhooks/machinepool.go

```diff
@@ -189,6 +189,14 @@ func (webhook *MachinePool) validate(oldObj, newObj *clusterv1.MachinePool) erro
 		}
 	}
 
+	if len(newObj.Spec.Template.Spec.Taints) > 0 {
+		if !feature.Gates.Enabled(feature.MachineTaintPropagation) {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("taints"), "taints are not allowed to be set when the feature gate MachineTaintPropagation is disabled"))
```

Comment:
This makes it sound like it would be allowed if the feature gate is enabled, maybe also mention that it's not implemented in this error message?